spec/paperclip/storage/fog_spec.rb

```diff
@@ -475,8 +506,44 @@ def custom_method
           assert_equal @dummy.avatar.fog_credentials, @dynamic_fog_credentials
         end
       end
-    end
 
+      context "with AES256 encryption" do
+        before do
+          rebuild_model(@options.merge(fog_encryption: :aes256))
+          @dummy = Dummy.new
+          @dummy.avatar = @file
+        end
+
+        it 'sets the @fog_encryption instance variable' do
+          assert_equal 'AES256', @dummy.avatar.instance_variable_get('@fog_encryption')
+        end
+
+        it "passes the fog_encryption type to the Fog::Storage::AWS::Files instance" do
+          Fog::Storage::AWS::Files.any_instance.expects(:create).with(has_entry(:encryption, 'AES256'))
+          @dummy.save
+        end
+      end
+
+      context "with fog_encryption disabled in multiple ways" do
+        [nil, false, ''].each do |tech|
+          before do
+            rebuild_model(@options.merge(fog_encryption: tech))
+            @dummy = Dummy.new
+            @dummy.avatar = @file
+          end
+
+          it 'sets the @fog_encryption instance variable' do
+            assert_equal false, @dummy.avatar.instance_variable_get('@fog_encryption')
```

Comment:
Line is too long. [86/80]<br>Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.